src/arrayMethodSort.js

```diff
@@ -4,8 +4,59 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const length = this.length;
+
+    let comparator;
+
+    if (typeof compareFunction !== 'function') {
```

Comment:
It's good that you've checked the type of `compareFunction` before using it. However, you might want to consider throwing an error if the `compareFunction` is neither a function nor `undefined`. This would make your method more robust and easier to debug.